topic1/TestShoppingCart.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package shoppingcart;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ *
+ * @author roberta
+ */
+public class TestShoppingCart {
+
+    private static List<Item> n;
+
+    public static void main(String[] args) {
+        Product tv = new Product("Tv led", "Resolution screen: 1366 x 768 pixels", 2500);
+        Product freezer = new Product("Freezer", "352 liters capacity", 8000);
+        Product fridge = new Product("Fridge Atma", "Stainless steel. No Frost", 9600);
+
+        Item item1 = new Item(1, tv);
+        Item item2 = new Item(1, freezer);
+        Item item3 = new Item(1, fridge);
+
+        ShoppingCart sc = new ShoppingCart();
+        List<Item> itemList = new ArrayList<Item>();
+        itemList.add(item1);
+        System.out.println(item1);
+        itemList.add(item2);
+        System.out.println(item2);
+        itemList.add(item3);
+        System.out.println(item3);
+        sc.setItemList(itemList);
+        sc.setPayment(new CreditCard(new Date(),36236541,"Visa",new Date()));
+        sc.setPayment(new PaypalPayment("user@name.com", "hola123",new Date()));
+        
+        User user = new User("Ana", "user@name.com", "123456");
+        MySystem.getInstance().addObserver(user);
+        
+        System.out.println(sc.getDiscountCreditCard());
+        System.out.println(sc.toString());
+        System.out.println(sc.getLessExpensiveItem());
+        
+        MySystem.getInstance().newSale(sc);
+        
```

Comment:
Define the expected and desired output for a normal case, with correct input.

    Now, implement the test by declaring a class, name it anything (Usually something like TestAddingModule), and add the testAdd method to it (i.e. like the one below) :
        Write a method , and above it add the @Test annotation.
        In the method, run your binary sum , and assertEquals(expectedVal,calculatedVal).

        Test your method by running it (i.e. in Eclipse, right click , run as, junit test).

        //for normal addition 
        @Test
        public void testAdd1Plus1() 
        {
            int x  = 1 ; int y = 1;
            assertEquals(2, myClass.add(x,y));
        }

    Add other cases as desired.
        Test that your binary sum does not throw a unexpected exception if there is an integer overflow.

        Test that your method handles Null inputs gracefully (for example).

        //if you are using 0 as default for null, make sure your class works in that case.
        @Test
        public void testAdd1Plus1() 
        {
            int y = 1;
            assertEquals(0, myClass.add(null,y);
        }